### PR TITLE
Update sns_aggregator IC cargo dependencies to release-2023-06-07_23-01

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
We would like to update our dependencies on the IC repository.

Updating all dependencies at once causes too many changes, so this is done piecemeal.

# Changes
## Resolver
```
0.271 warning: some crates are on edition 2021 which defaults to `resolver = "2"`, but virtual workspaces default to `resolver = "1"`                                                                                                                                                                                                                                                                                                                                                                                                                                          
0.271 note: to keep the current resolver, specify `workspace.resolver = "1"` in the workspace root's manifest                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
0.271 note: to use the edition 2021 resolver, specify `workspace.resolver = "2"` in the workspace root's manifest 
...
66.97 error[E0432]: unresolved import `crate::sys::IoSourceState`
```
Addresed by:
```
 [workspace]
+resolver = "2"
```
## Update IC commit
Update the `IC` repo commit used in the `sns_aggregator` `Cargo.toml` to the latest release that works without any further code changes.

# Tests
See CI.

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

File: CHANGELOG-Sns_Aggregator.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ The SNS Aggregator is released through proposals in the Network Nervous System.
 
 ### Added
 ### Changed
+* Updated libraries provided by the `ic` repository.
```
Comment (lmuntaner) [inline CHANGELOG-Sns_Aggregator.md:13]: Maybe mention to which release they were updated?

**Review by lmuntaner — APPROVED** [review]
The SNS Aggregator also has dependency on rust crates from the IC also.

Good to know, I guess it's one more dependency to manage 🙈

File: CHANGELOG-Sns_Aggregator.md (hunk 1 above)
Comment (bitdivine) [reply CHANGELOG-Sns_Aggregator.md:13]: I was being intentionally vague as I mean to keep pushing this with a sequence of PRs but, sure, I could.

Comment (lmuntaner) [reply CHANGELOG-Sns_Aggregator.md:13]: ah, ok, in that case ok. So the changelog will cover all the upcoming. Maybe change it in the last PR. Or in each PR if you prefer. But no need to have an entry per update.

**Comment by bitdivine** [comment]
@lmuntaner Could you re-approve, please?  GitHub isn't working in this repo as it did in tests on the snsdemo repo.

**Review by lmuntaner — APPROVED** [review]

**Comment by bitdivine** [comment]
Note: The patch-update also makes this change in the lockfile:
```
- "ic-nns-constants",
+ "ic-nns-constants 0.8.0 (git+https://github.com/dfinity/ic?rev=89129b8212791d7e05cab62ff08eece2888a86e0)",
```
Once that change is made, though, future updates are stable.

**Comment by bitdivine** [comment]
Can you re-approve, please, @lmuntaner .  I had the temerity to merge in main so need another approval.

**Review by lmuntaner — APPROVED** [review]

**Pull request merged**